Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -3621,11 +3653,22 @@ private unsafe void OnUpdate(InputUpdateType updateType, ref InputEventBuffer ev
                         break;
                 }
 
-                m_Metrics.totalEventProcessingTime +=
+                ResetCurrentProcessedEventBytesForDevices();
+
+                // Update metrics (exposed via analytics and debugger)
+                var eventProcessingTime =
                     ((double)(Stopwatch.GetTimestamp() - processingStartTime)) / Stopwatch.Frequency;
+                m_Metrics.totalEventCount += totalEventCount;
+                m_Metrics.totalEventBytes += totalEventSizeBytes;
+                m_Metrics.totalEventProcessingTime += eventProcessingTime;
                 m_Metrics.totalEventLagTime += totalEventLag;
 
-                ResetCurrentProcessedEventBytesForDevices();
+                // Profiler counters
+                InputStatistics.EventCount.Value += totalEventCount;
+                InputStatistics.EventSize.Value += totalEventSizeBytes;
+                InputStatistics.AverageLatency.Value += ((totalEventLag / totalEventCount) * 1e9);
+                InputStatistics.MaxLatency.Value += (maxEventLag * 1e9);
+                InputStatistics.EventProcessingTime.Value += eventProcessingTime * 1e9; // TODO Possible to replace Stopwatch with marker somehow?
```

Comment:
I guess this last line here is an anti pattern, ideally I would like to include the Update profiler marker into the profiler module, is this possible?

Reply:
During hackweek I used StopWatch.  It was preallocated and reset and reused every frame and accumulated in a [ProfilerCounter](https://docs.unity3d.com/Packages/com.unity.profiling.core@0.2/manual/profilercounter-guide.html)

Reply:
which is what it looks like you're doing here :)